### DISCUSSION TITLE
cant manage to convert a string to nlohmann::json

### DISCUSSION THREAD

**Discussion by GodGamer1** [nlohmann/json#3509] [Q&A]
Hi. i am trying to convert a string to  json i have looked for solutions and none of them worked for me.
rn i have this code with this: 

std::cout << value << std::endl;
nlohmann::json j = nlohmann::json::parse(value);
std::cout <<"check1" << j.begin().value() << "check2" << std::endl;

it never arrives to check1

**Comment by GodGamer1** [comment]
forgot to say. value is a string that i did json.dump() into it

**Comment by falbrechtskirchinger** [comment]
What is the content of `value`? `nlohmann::json::parse()` is probably throwing an exception, can you catch it and output `.what()`?

**Reply by GodGamer1** [reply]
i dont think .what() will work because the code gets stuck after that line

**Reply by falbrechtskirchinger** [reply]
```cpp
try {
    std::cout << value << std::endl;
    nlohmann::json j = nlohmann::json::parse(value);
    std::cout <<"check1" << j.begin().value() << "check2" << std::endl;
}
catch(const nlohmann::json::parse_error &ex) {
    std::cout << ex.what() << "\n";
}
```

**Comment by GodGamer1** [comment]
value is {username:asdfsdf,password:sdfgsdg} for example

**Reply by falbrechtskirchinger** [reply]
Do you mean
```json
{"username": "asdfsdf", "password": "sdfgsdg"}
```
?

Because what you wrote isn't valid JSON.

**Reply by GodGamer1** [reply]
thats the string whenever i print it 
{username:asdfsdf,password:sdfgsdg}

**Reply by falbrechtskirchinger** [reply]
Well, that's not JSON. That's why it's not working.

**Reply by falbrechtskirchinger** [reply]
You're missing quotes. This can't be the verbatim output of `json.dump()`.

**Reply by GodGamer1** [reply]
oh thx i understand it fixed my problem 
thx

**Reply by gregmarr** [reply]
If you are all set, can you please mark one of the replies as the answer?
